perf-testing/test_secretsmanager.rb

```diff
@@ -0,0 +1,367 @@
+MEASUREMENTS = ['Total request time (ms)', 'Serialization time (ms)', 'Deserialization time (ms)',
+                'Request payload size (bytes)', 'Response payload size (bytes)'].freeze
+
+def generate_create_secret_request(timestamp, iteration)
+  {
+    name: "TestSecret_#{timestamp}_#{iteration}",
+    secret_string: 'A temporary secret value',
+    description: "The testing secret for run #{iteration}",
+    tags: [
+      {
+        key: 'Stage',
+        value: 'Production'
+      },
+      {
+        key: 'Iteration',
+        value: iteration
+      }
+    ]
+  }
+end
+
+def generate_binary_create_secret_request(timestamp, iteration)
+  {
+    name: "TestBinarySecret_#{timestamp}_#{iteration}",
+    secret_string: 'A temporary secret value',
+    description: "The binary testing secret for run #{iteration}",
+    tags: [
+      {
+        key: 'Stage',
+        value: 'Production'
+      },
+      {
+        key: 'Iteration',
+        value: iteration
+      }
+    ]
+  }
+end
+
+def generate_put_secret_value_request(timestamp, iteration, size)
+  {
+    secret_id: "TestSecret_#{timestamp}_#{iteration}",
+    secret_string: (0...size).map { ASCII[rand(ASCII.length)] }.join
+  }
+end
+
+def generate_binary_put_secret_value_request(timestamp, iteration, size)
+  {
+    secret_id: "TestBinarySecret_#{timestamp}_#{iteration}",
+    secret_binary: Random.bytes(size)
+  }
+end
+
+def generate_get_secret_value_request(timestamp, iteration)
+  {
+    secret_id: "TestSecret_#{timestamp}_#{iteration}"
+  }
+end
+
+def generate_binary_get_secret_value_request(timestamp, iteration)
+  {
+    secret_id: "TestBinarySecret_#{timestamp}_#{iteration}"
+  }
+end
+
+def generate_describe_secret_request(timestamp, iteration)
+  {
+    secret_id: "TestSecret_#{timestamp}_#{iteration}"
+  }
+end
+
+def generate_list_secrets_request(iteration)
+  {
+    filters: [
+      {
+        key: 'tag-key',
+        values: ['Iteration']
+      },
+      {
+        key: 'tag-value',
+        values: [iteration]
+      }
+    ]
+  }
+end
+
+def generate_delete_secret_request(timestamp, iteration)
+  {
+    secret_id: "TestSecret_#{timestamp}_#{iteration}",
+    force_delete_without_recovery: true
+  }
+end
+
+def generate_binary_delete_secret_request(timestamp, iteration)
+  {
+    secret_id: "TestBinarySecret_#{timestamp}_#{iteration}",
+    force_delete_without_recovery: true
+  }
+end
+
+def separate_serde_data(data)
+  separated = Array.new(2) { [] }
+  data.each do |i|
+    separated[0] << i[0]
+    separated[1] << i[1]
+  end
+  separated
+end
+
+def separate_byte_data(data)
+  separated = Array.new(2) { [] }
+  data.each_with_index do |i, idx|
+    if idx.even?
+      separated[0] << i
+    else
+      separated[1] << i
+    end
+  end
+  separated
+end
+
+def write_test_output(operation, protocol, dimension, metric, input, outfile)
+  input.sort!
+  result = {
+    "service": 'SecretsManager',
```

Comment:
Ruby has weird syntax for Hashes.  What you're doing here is actually creating a symbol key and doesn't require you to use quotes. eg:
```
result = {
  service: 'SecretsManager'
...
}
```